piptools/scripts/sync.py

```diff
@@ -55,6 +55,10 @@
     is_flag=True,
     help="Ignore package index (only looking at --find-links URLs instead)",
 )
+@click.option(
+    "--python-executable",
```

Comment:
We could use `type=click.Path(exists=True)` instead of this check below:

```python
    if python_executable is not None and not os.path.exists(python_executable):
         msg = "Python executable {} not found"
         log.error(msg.format(python_executable))
         sys.exit(2)
```

Reply:
Great idea, thanks!